src/box/lua/net_box.lua

```diff
@@ -390,6 +391,49 @@ local function new_sm(uri, opts)
                     })
                 end
             end
+            if not opts.fetch_schema then
+                remote.space = setmetatable({}, {
+                    __index = function(_, space_key)
+                        local id_or_name
+                        if type(space_key) == 'number' then
+                            id_or_name = 'id'
+                        elseif type(space_key) == 'string' then
+                            if not features.space_and_index_names then
+                                return nil
+                            end
+                            id_or_name = 'name'
+                        else
+                            return nil
+                        end
+                        local space = {[id_or_name] = space_key,
+                                       _id_or_name = space_key}
+                        space.index = setmetatable({}, {
+                            __index = function(self, idx_key)
```

Comment:
Nit: `self` is unused. Should be replaced with `_`.

Reply:
Fixed.